### PR TITLE
Handle SSH failure gracefully during the installation process

### LINKED ISSUE #1947: Installer to handle ssh failure gracefully
If there is an ssh problem, we can output a message and skip the initialization step instead of kept on re trying.

### PR DESCRIPTION
**Pull request by joecheuk**
Resolves #1947

### CODE REVIEW

File: install/gcp/installer/util/gcloud.py [hunk 1]
```diff
@@ -803,13 +805,20 @@ def check_vm_init_status(vm_name, zone):
...
         ['gcloud', 'compute', 'ssh', vm_name,
          '--zone', zone, '--command', check_script_executed, '--quiet'])
     # --quiet flag is needed to eliminate the prompting for user input
     # which will hang the run_command function
     # i.e. It will create a folder at ~/.ssh and generate a new ssh key
 
+    if err:
+        print('Error occurred when sshing to the VM, will skip waiting for '
+              'the instance to be initialized and proceed with the rest of '
+              'the installation.')
```
Comment (blueandgold) [inline install/gcp/installer/util/gcloud.py]: I would add a message to tell the user that they might need to verify the VM is actually initialized themselves.

```
'Error occurred when sshing to the VM.  Unable to verify the VM is fully initialized.'
'This is a non-fatal error, but you should verify the VM is fully initialized before use (add how).'
'Will skip waiting for the instance to be initialized and proceed with the rest of '
'the installation.'
```

File: install/gcp/installer/forseti_installer.py [hunk 2]
```diff
@@ -307,10 +308,14 @@ def wait_until_vm_initialized(self, vm_name):
...
-            max_loading_time=constants.MAXIMUM_LOADING_TIME_IN_SECONDS,
-            exit_condition_checker=status_checker,
-            message=loading_message)
+        try:
+            _ = utils.start_loading(
+                max_loading_time=constants.MAXIMUM_LOADING_TIME_IN_SECONDS,
+                exit_condition_checker=status_checker,
+                message=loading_message)
+        except installer_errors.SSHError:
+            # Expected error, don't need to do anything.
```
Comment (blueandgold) [inline install/gcp/installer/forseti_installer.py]: It is good to have a comment here.  But I would clarify why this is an expected error, and why we don't need to do anything.

File: install/gcp/installer/util/gcloud.py [hunk 3]
```diff
@@ -803,13 +805,20 @@ def check_vm_init_status(vm_name, zone):
...
-    _, out, _ = utils.run_command(
+    _, out, err = utils.run_command(
         ['gcloud', 'compute', 'ssh', vm_name,
          '--zone', zone, '--command', check_script_executed, '--quiet'])
     # --quiet flag is needed to eliminate the prompting for user input
     # which will hang the run_command function
     # i.e. It will create a folder at ~/.ssh and generate a new ssh key
 
+    if err:
+        print('Error occurred when sshing to the VM, will skip waiting for '
```
Comment (blueandgold) [inline install/gcp/installer/util/gcloud.py]: Should this be in constants?

File: install/gcp/installer/forseti_installer.py (hunk 2 above)
Comment (joecheuk) [reply install/gcp/installer/forseti_installer.py]: Updated.

File: install/gcp/installer/util/gcloud.py (hunk 3 above)
Comment (joecheuk) [reply install/gcp/installer/util/gcloud.py]: Updated.

File: install/gcp/installer/util/gcloud.py (hunk 1 above)
Comment (joecheuk) [reply install/gcp/installer/util/gcloud.py]: Updated.

**Pull request merged**